packages/firestore/src/model/mutation.ts

```diff
@@ -365,12 +386,47 @@ export function extractMutationBaseValue(
   mutation: Mutation,
```

Comment:
Should this `mutation` be a TranformMutation? It might be cleaner to move the type check to the callsite. You can probably then inline the inner function again (since the method is now simpler).